admin/admin.go

```diff
@@ -50,6 +73,22 @@ func (s *AdminServer) Start(ctx context.Context) error {
 		}
 	}()
 
+	// Start RPC server in a goroutine
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+		http.Handle("/", rpcServer)
+		err := http.ListenAndServe(fmt.Sprintf(":%d", s.port+1), nil)
```

Comment:
the rpc server should just be on the same port as the specified

there should be very limited direct http endpoints here so 

`/`: defaults to rpc
`/ready`: probably the only http endpoint we need

these can all be on the same port since they are separated by the route

Reply:
awesome got it, making these revisions